cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -145,6 +150,13 @@ func (dp *devicePlugin) scan() (dpapi.DeviceTree, error) {
 				ContainerPath: devPath,
 				Permissions:   "rw",
 			})
+			klog.V(4).Infof("Adding %s to GPU %s/%s", devPath, monitorType, monitorID)
+			// querying metrics requires device to be also writable
+			monitor = append(monitor, pluginapi.DeviceSpec{
+				HostPath:      devPath,
+				ContainerPath: devPath,
+				Permissions:   "rw",
+			})
```

Comment:
Nit: probably makes sense to emphasize the fact that `nodes` contains a subset of `monitor` by instantiating a device spec once and appending its copies to `nodes` and `monitor`, e.g.
```go
devSpec := pluginapi.DeviceSpec{
        HostPath:      devPath,
        ContainerPath: devPath,
        Permissions:   "rw",
}
nodes = append(nodes, devSpec)
monitor = append(monitor, devSpec)
```

Reply:
Good idea, that makes the code nicer -> pushed update